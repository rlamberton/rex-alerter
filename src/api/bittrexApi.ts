// CORS proxy is required since Bittrex does not allow AJAX from browser
//const CORS_PROXY = 'https://cors-anywhere.herokuapp.com/';
const CORS_PROXY = 'http://localhost:8080/';
const MARKET_TICKERS_URL = 'https://api.bittrex.com/v3/markets/tickers';
const MARKET_SUMMARIES_URL = 'https://api.bittrex.com/v3/markets/summaries';
const MARKETS_URL = 'https://api.bittrex.com/v3/markets';
const CURRENCIES_URL = 'https://api.bittrex.com/v3/currencies';

async function getMarketTickers() {
    const response = await fetch(CORS_PROXY + MARKET_TICKERS_URL);
    if (response.ok) {
        const json = await response.json();
        return json;
    }

    return response;
}

async function getMarketSummaries() {
    const response = await fetch(CORS_PROXY + MARKET_SUMMARIES_URL);
    if (response.ok) {
        const json = await response.json();
        return json;
    }

    return response;
}

async function getMarkets() {
    const response = await fetch(CORS_PROXY + MARKETS_URL);
    if (response.ok) {
        const json = await response.json();
        return json;
    }

    return response;
}

async function getCurrencies() {
    const response = await fetch(CORS_PROXY + CURRENCIES_URL);
    if (response.ok) {
        const json = await response.json();
        return json;
    }

    return response;
}

export { getMarketTickers, getMarketSummaries, getMarkets, getCurrencies };
