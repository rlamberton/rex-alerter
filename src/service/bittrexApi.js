// CORS proxy is required since Bittrex does not allow AJAX from browser
//const CORS_PROXY = 'https://cors-anywhere.herokuapp.com/';
const CORS_PROXY = 'http://localhost:8080/';
const MARKET_TICKERS_URL = 'https://api.bittrex.com/v3/markets/tickers';
const MARKET_SUMMARIES_URL = 'https://api.bittrex.com/v3/markets/summaries';
const MARKETS_URL = 'https://api.bittrex.com/v3/markets';

async function getMarketTickers() {
    const response = await fetch(CORS_PROXY + MARKET_TICKERS_URL, {'Content-Type': 'application/json', 'Accept': 'application/json'});
    if (response.ok) {
        const json = await response.json();
        return json;
    }

    return response;
}

async function getMarketSummaries() {
    const response = await fetch(CORS_PROXY + MARKET_SUMMARIES_URL, {'Content-Type': 'application/json', 'Accept': 'application/json'});
    if (response.ok) {
        const json = await response.json();
        return json;
    }

    return response;
}

async function getMarkets() {
    const response = await fetch(CORS_PROXY + MARKETS_URL, {'Content-Type': 'application/json', 'Accept': 'application/json'});
    if (response.ok) {
        const json = await response.json();
        return json;
    }

    return response;
}

export { getMarketTickers, getMarketSummaries, getMarkets };
