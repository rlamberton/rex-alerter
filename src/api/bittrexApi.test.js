import { getMarketTickers, getMarketSummaries, getMarkets, getCurrencies } from './bittrexApi';

test('check getMarketTickers', () => {
    const results = getMarketTickers();
    expect( results ).toMatchSnapshot();
});

test('check getMarketSummaries', () => {
    const results = getMarketSummaries();
    expect( results ).toMatchSnapshot();
});

test('check getMarkets', () => {
    const results = getMarkets();
    expect( results ).toMatchSnapshot();
});

test('check getCurrencies', () => {
    const results = getCurrencies();
    expect( results ).toMatchSnapshot();
});