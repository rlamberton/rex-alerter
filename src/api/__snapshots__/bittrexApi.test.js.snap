// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check getCurrencies 1`] = `Promise {}`;

exports[`check getMarketSummaries 1`] = `Promise {}`;

exports[`check getMarketTickers 1`] = `Promise {}`;

exports[`check getMarkets 1`] = `Promise {}`;
